src/components/EditPartnership/EditPartnership.tsx

```diff
@@ -0,0 +1,112 @@
+import { Box, Button } from '@material-ui/core';
+import { useSnackbar } from 'notistack';
+import React, { FC } from 'react';
+import { PartnershipType, UpdatePartnershipInput } from '../../api';
+import { DialogForm } from '../Dialog/DialogForm';
+import { DialogState } from '../Dialog/useDialog';
+import {
+  CheckboxesField,
+  CheckboxOption,
+  RadioField,
+  RadioOption,
+  SubmitError,
+} from '../form';
+import { PartnershipCardFragment } from '../PartnershipCard/PartnershipCard.generated';
+import {
+  useDeletePartnershipMutation,
+  useUpdatePartnershipMutation,
+} from './EditPartnership.generated';
+
+interface EditPartnershipProps {
+  state: DialogState;
+  partnership: PartnershipCardFragment;
+}
+
+export const EditPartnership: FC<EditPartnershipProps> = ({
+  state,
+  partnership,
+}) => {
+  const [updatePartnership] = useUpdatePartnershipMutation();
+  const [deletePartnership] = useDeletePartnershipMutation();
+  const { enqueueSnackbar } = useSnackbar();
+
+  const submit = async (input: UpdatePartnershipInput) => {
+    const inputWithId: UpdatePartnershipInput = {
+      partnership: { ...input.partnership, id: partnership.id },
+    };
+    await updatePartnership({
+      variables: { input: inputWithId },
+      refetchQueries: ['ProjectPartnerships'],
+    });
+    enqueueSnackbar(
+      `Updated Partnership with ${partnership.organization.name.value}`,
+      {
+        variant: 'success',
+      }
+    );
+  };
+
+  const onDeleteClick = async () => {
+    const { id } = partnership;
+    await deletePartnership({
+      variables: { input: id },
+      refetchQueries: ['ProjectPartnerships'],
+    });
+    state.onClose();
+    enqueueSnackbar(
+      `Deleted Partnership with ${partnership.organization.name.value}`,
+      {
+        variant: 'success',
+      }
+    );
+  };
+
+  // TODO: have DialogForm pass down initialValues to the radio options (currently passed to Checkboxes)
+  // const initialValues = {
+  //   partnership: {
+  //     id: partnership.id,
+  //     agreementStatus: partnership.agreementStatus.value,
+  //     mouStatus: partnership.mouStatus.value,
+  //     types: partnership.types.value,
+  //   },
+  // };
+
+  return (
+    <DialogForm
+      onSubmit={submit}
+      {...state}
+      DialogProps={{
+        fullWidth: true,
+        maxWidth: 'xs',
+      }}
+      title="Edit Partnership"
```

Comment:
Oh it might be good to include the org name in this title

Reply:
added